modules/git/signature.go

```diff
@@ -4,7 +4,46 @@
 
 package git
 
-const (
-	// GitTimeLayout is the (default) time layout used by git.
-	GitTimeLayout = "Mon Jan _2 15:04:05 2006 -0700"
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
 )
+
+// Helper to get a signature from the commit line, which looks like:
+//
+//	full name <user@example.com> 1378823654 +0200
+//
+// Haven't found the official reference for the standard format yet.
+// This function never fails, if the "line" can't be parsed, it returns a default Signature with "zero" time.
+func parseSignatureFromCommitLine(line string) *Signature {
+	sig := &Signature{}
+	s1, sx, ok1 := strings.Cut(line, " <")
+	s2, s3, ok2 := strings.Cut(sx, "> ")
+	if !ok1 || !ok2 {
+		sig.Name = line
+		return sig
+	}
+	sig.Name, sig.Email = s1, s2
+
+	if strings.Count(s3, " ") == 1 {
+		ts, tz, _ := strings.Cut(s3, " ")
+		seconds, _ := strconv.ParseInt(ts, 10, 64)
+		if tzTime, err := time.Parse("-0700", tz); err == nil {
+			sig.When = time.Unix(seconds, 0).In(tzTime.Location())
+		}
+	} else {
+		// the old gitea code tried to parse the date in a few different formats, but it's not clear why.
+		// according to public document, only the standard format "timestamp timezone" could be found, so drop other formats.
```

Comment:
It seems that only `xxxxdate` can be set as other formats, but we use `creator` here which seems only using `git internal format`.
https://github.com/git/git/blob/235986be822c9f8689be2e9a0b7804d0b1b6d821/Documentation/git-for-each-ref.txt#L362-L364
All formats can be found here:
https://git-scm.com/docs/git-rev-list#Documentation/git-rev-list.txt---dateltformatgt

Reply:
I do not see how these links are related .... the documents describe about the behaviors of some commands, but not the "internal data structure".

Reply:
The only possible clue in these documents is `--date=raw`, the `raw` might mean the "raw data (time) save in a commit object"